packages/vite-node/src/client.ts

```diff
@@ -61,23 +61,23 @@ export class ModuleCacheMap extends Map<string, ModuleCache> {
   update(fsPath: string, mod: Partial<ModuleCache>) {
     fsPath = this.normalizePath(fsPath)
     if (!super.has(fsPath))
-      super.set(fsPath, mod)
+      this.setByModuleId(fsPath, mod)
     else
       Object.assign(super.get(fsPath) as ModuleCache, mod)
     return this
   }
 
-  setByModuleId(modulePath: string, mod: ModuleCache) {
-    return super.set(modulePath, mod)
+  setByModuleId(modulePath: string, mod: Partial<ModuleCache>) {
+    return super.set(modulePath, { imports: new Set(), importers: new Set(), ...mod })
```

Comment:
Can we not create a new object here?

Reply:
Maybe instead of `.set({})` inside of `.get()` we can just provide `{ imports: new Set(), importers: new Set() }`?